### PR TITLE
Working MNIST LeNet

### PR DESCRIPTION
**Pull request by wutayng**
Pieced together a working MNIST trainer - 9797/10000 (98%) accuracy on 10000samp MNIST test set after 1 epoch - just running on local CPU at this point!

### CODE REVIEW

File: lth_reproduction/src/mnist.py [hunk 1]
```diff
@@ -1,2 +1,124 @@
...
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torchvision import datasets, transforms
+from torch.optim.lr_scheduler import StepLR
+import gin
+
+
+class LeNet(nn.Module):
```
Comment (wutayng) [inline lth_reproduction/src/mnist.py:15]: Put together this LeNet using some existing sources and the LTH paper - LTH specifically mentions in figure 2 that the convolutions are 3x3 and fc layers are 300,100,10 - so I put that in, but I am a little unsure if the rest of the architecture (convolution blocks in particular) line up with LTH's LeNet

**Comment by grantgasser** [comment]
Awesome dude! How long did the epoch take? How long did prediction take? Can you write the results to a file? Or maybe we could move the code to a jupyter notebook to show some visualizations? Thx for doing the requirements and config files too!
